### PR TITLE
Journalist can categorize Articles depending on topic

### PR DESCRIPTION
**Pull request by AlexSaxena**
[PT-link](https://www.pivotaltracker.com/story/show/169253782)

## PR content
In this PR we are adding a feature where a journalist can choose a category for given article.

## This PR contains:
- Unit tests for model and factories
- Request spec tests for including categories
- Seeded Categories

## What have we learned?
- Improved understanding of Rails and Associations 
- Improved understanding of RSpec and related technologies

### CODE REVIEW

File: app/controllers/v1/articles_controller.rb [hunk 1]
```diff
@@ -21,8 +21,13 @@ def show
   end
 
   def create
+    
+
```
Comment (clalil) [inline app/controllers/v1/articles_controller.rb]: Are these spaces in the actual file as well or is it just Github magic?

**Review by shsteindorsson — APPROVED** [review]

**Review by clalil — APPROVED** [review]

File: spec/models/category_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
```
Comment (faraznaeem) [inline spec/models/category_spec.rb:1]: no test for the association's

**Review by faraznaeem — CHANGES_REQUESTED** [review]

**Review by lucamarial — APPROVED** [review]

**Review by jonas-masharqa — APPROVED** [review]

File: spec/models/category_spec.rb (hunk 2 above)
Comment (GergKllai1) [reply spec/models/category_spec.rb:1]: Still no tests for assications

File: spec/requests/v1/api_provides_possibility_for_journalist_to_create_article_spec.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 RSpec.describe 'Can create article with attributes' do
   let(:journalist) { create(:user, role: 'journalist') }
+  let(:category) {FactoryBot.create(:category)}
```
Comment (GergKllai1) [inline spec/requests/v1/api_provides_possibility_for_journalist_to_create_article_spec.rb]: You don't need FactoryBot, remove it for consistency

File: db/seeds.rb [hunk 4]
```diff
@@ -1,2 +1,72 @@
-users = User.create(email: 'user@mail.com', password: 'password')
+User.destroy_all
+
+user = User.create(email: 'user@mail.com', encrypted_password: 'password', role: 'user')
```
Comment (GergKllai1) [inline db/seeds.rb]: it is just password, for consistency change it pls

**Review by GergKllai1 — CHANGES_REQUESTED** [review]

**Review by GergKllai1 — APPROVED** [review]

**Pull request merged**